main.go

```diff
@@ -81,21 +95,52 @@ func serverFinalizer(server *Server) {
 //
 // FIXME Pass in a list of 8-bit tags defining epochs.
 // The instance will generate its own secret key.
-func NewServer() (*Server, error) {
+func NewServer() (*Server, *string, error) {
 	// FIXME should we runtime.LockOSThread() here?
 	raw := C.randomness_server_create()
 	if raw == nil {
-		return nil, errors.New("failed to create randomness server")
+		return nil, nil, errors.New("failed to create randomness server")
 	}
 	server := &Server{raw: raw}
 	runtime.SetFinalizer(server, serverFinalizer)
-	return server, nil
+
+	var pkOutput [serializedPkBufferSize]byte
+	pkSize := C.randomness_server_get_public_key(
+		server.raw, (*C.uint8_t)(unsafe.Pointer(&pkOutput[0])))
+	if pkSize == 0 {
+		return nil, nil, errors.New("failed to get public key")
+	}
+	pk := base64.StdEncoding.EncodeToString(pkOutput[:pkSize])
+
+	return server, &pk, nil
 }
 
-// getEpoch returns the ISO week number of the given timestamp.
-func getEpoch(t time.Time) epoch {
-	_, week := t.ISOWeek()
-	return epoch(week)
+// getEpoch returns the current epoch for the given timestamp.
+func getEpoch(firstEpochTime time.Time, refTime time.Time) (epoch, string) {
+	currentSecondsSinceFirstEpoch := refTime.Unix() - firstEpochTime.Unix()
+	epochsSinceFirstEpoch := currentSecondsSinceFirstEpoch / epochLengthTimeSec
+	nextEpochTime := time.Unix(firstEpochTime.Unix()+
+		(epochLengthTimeSec*(epochsSinceFirstEpoch+1)), 0)
+	nextEpochTime = nextEpochTime.In(time.UTC)
+	currentEpoch := epochsSinceFirstEpoch % 256
+	return epoch(currentEpoch), nextEpochTime.Format(time.RFC3339)
+}
+
+// getServerInfo returns the current epoch info and public key.
+func getServerInfo(pk string) http.HandlerFunc {
+	firstEpochTime, _ := time.Parse(time.RFC3339, firstEpochTimestamp)
+	return func(w http.ResponseWriter, r *http.Request) {
+		currentEpoch, nextEpochTime := getEpoch(firstEpochTime, time.Now())
+		resp := serverInfoResponse{
+			PublicKey:     pk,
+			CurrentEpoch:  currentEpoch,
+			NextEpochTime: nextEpochTime,
```

Comment:
Would it be cleaner to have `getEpoch` return `nextEpochTime` as a `time.Time` and convert to RFC3339 here? That way all the formatting is in one place

Reply:
Sure, I think that makes sense considering `getEpoch` also takes `time.Time` objects as parameters